core/src/main/java/net/opentsdb/aura/metrics/core/gorilla/GorillaTimeSeriesDownSamplingEncoder.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * This file is part of OpenTSDB.
+ * Copyright (C) 2021  Yahoo.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.opentsdb.aura.metrics.core.gorilla;
+
+import net.opentsdb.aura.metrics.core.downsample.DownSampler;
+import net.opentsdb.aura.metrics.core.downsample.DownSamplingTimeSeriesEncoder;
+import net.opentsdb.aura.metrics.core.downsample.Interval;
+import net.opentsdb.aura.metrics.core.downsample.SegmentWidth;
+
+import java.util.Arrays;
+import java.util.Iterator;
+
+public class GorillaTimeSeriesDownSamplingEncoder
+    extends BaseGorillaSegmentEncoder<OffHeapDownSampledGorillaSegment>
+    implements DownSamplingTimeSeriesEncoder {
+
+  private short intervalCount;
+  private Interval interval;
+  private SegmentWidth segmentSize;
+  private DownSampler downSampler;
+  private byte aggId;
+
+  public GorillaTimeSeriesDownSamplingEncoder(
```

Comment:
We also need the user to define the aggs they want. 
And we need a way to serialize the different aggs to different locations in an efficient manner without having to reload all of the bits from memory if we can.

Reply:
work in progress

Reply:
The serde to aerospike is work in progress.